tcrconvert/convert.py

```diff
@@ -175,3 +177,55 @@ def convert_gene(df, frm, to, species='human', frm_cols=[], quiet=False):
     out_df = out_df.replace('NoData', pd.NA).fillna(pd.NA)
 
     return out_df
+
+
+# Command-line version of convert_gene()
+@click.command(name='convert-gene', no_args_is_help=True)
+@click.option('-i', '--infile', help='Path to input CSV or TSV', required=True)
+@click.option('-o', '--outfile', help='Path to output CSV or TSV', required=True)
+@click.option('-f', '--frm', help='Input format of TCR data', required=True,
+              type=click.Choice(['tenx', 'adaptive', 'adaptivev2', 'imgt'], case_sensitive=False))
+@click.option('-t', '--to', help='Output format of TCR data', required=True,
+              type=click.Choice(['tenx', 'adaptive', 'adaptivev2', 'imgt'], case_sensitive=False))
+@click.option('-s', '--species', default='human', help="Species folder name under 'tcrconvert/data/'", show_default=True)
+@click.option('-c', '--frm_cols', default=[], help='List of custom V/D/J/C gene column names.', show_default=True,
+              multiple=True)
+@click.option('-q', '--quiet', is_flag=True, default=False, help='Whether to suppress warning messages.', show_default=True)
+def convert_gene_cli(infile, outfile, frm, to, species, frm_cols, quiet):
+    '''Convert T-cell receptor V/D/J/C gene names.
+
+    :Example:
+
+    Example uisng custom input columns 'myV', 'myD', 'myJ'
+
+    .. code-block:: bash
+
+       $ tcrconvert convert-gene --infile 10x_tcrs.csv --outfile converted.tsv --frm tenx --to adaptive --species mouse -c myV -c myD -c myJ --quiet
+    '''
+
+    # Check that input and output paths are CSV/TSV
+    if not infile.endswith(('csv', 'tsv')):
+        logger.error('"infile" must be a .csv or .tsv file')
+        raise(ValueError)
+
+    if not outfile.endswith(('csv', 'tsv')):
+        logger.error('"outfile" must be a .csv or .tsv file')
+        raise(ValueError)
+
+    # Load data
+    # For our purposes, read in every column as string so that boolean values
+    # don't get converted from uppercase to capitalized, etc.
+    if infile.endswith('csv'):
+        df = pd.read_csv(infile, dtype=str)
+    elif infile.endswith('tsv'):
+        df = pd.read_csv(infile, sep='\t', dtype=str)
+
+    # Convert gene names
+    # Cast frm_cols as list because will be read in from command line as tuple
+    out_df = convert_gene(df, frm, to, species, list(frm_cols), quiet)
+
+    # Save output
+    if outfile.endswith('csv'):
+        out_df.to_csv(outfile, index=False)
+    elif outfile.endswith('tsv'):
+        out_df.to_csv(outfile, sep='\t', index=False)
```

Comment:
might want to `click.echo()` a message that the file has been written to a the path that's declared by `outfile`

Reply:
- Added this as part of printed info to CLI users with `click.echo` in 294aeba1c7dac5a25a991449e00c6a90392c7d39